### ISSUE TITLE
Buttons with links not clickable in Facebook email

### ISSUE THREAD

**Issue by louim** [Foundry376/Mailspring#301]
##### Are there any related issues?

No

##### What operating system are you using?

Mac OS X Sierra

##### What version of Mailspring are you using?
1.0.7
--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?

No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?

No

##### Is the issue reproducible with a particular attachment, message, signature, etc?

Received a "new connection email" from Facebook. Links at the bottom (red square) are not clickable in Mailspring. Confirmed working on mail.google.com

<img width="555" alt="capture d ecran 2017-11-03 a 11 16 26" src="https://user-images.githubusercontent.com/923718/32381386-ed266f5e-c088-11e7-8f1b-6a29a88dc316.png">

Here is the [body of the message](https://gist.github.com/louim/fb4cd6dedbabe3c0bcbee259c31d58aa#file-body-html-L222-L238). I redacted the mail and some of the Facebook params, so the links may not be valid anymore.

--

**Feature Request?**

**Comment by bengotow** [comment]
Hey thanks for filing this—this is super interesting. Did you pull this HTML from the Mailspring UI or via "Show Original" on Gmail? I /think/ what's happening is that Mailspring is corrupting the URLs similar to the way it was breaking image src attributes in some emails in #260. Could you try toggling the Developer Tools in Mailspring, and using "Inspect Element" to see what the HTML looks like inside the email iframe? I think the `href` on those links must be broken in some way (maybe a leading space character?)

**Comment by louim** [comment]
Hey!
The html in the gist is coming from Mailspring, via the `log data button` and `window.__message.body`. here is the relevant part from the "show original email" from gmail.

**Gmail, via "show original email", and decoding `quoted-printable`:**

```
<td class="mb_blk" style="">
    <a href="https://www.facebook.com/n/?login_alerts%2Fsettings%2F&amp;medium=email&amp;mid=55d15436cG2bf&amp;bcode=2.0964.AbyhsJFgi0Pjg&amp;
n_m=redacted%40gmail.com" style="color:#3b5998;text-decoration:none;">
        <table border="0" width="100%" cellspacing="0" cellpadding="0" style="border-collapse:collapse;">
            <tr>
                <td style="border-collapse:collapse;border-radius:2px;text-align:center;display:block;border:solid 1px #c9ccd1;box-shadow:0 1px 0 rgba(0, 0, 0, 0.04);background:#f6f7f8;padding:7px 16px 11px 16px;">
                    <a href=""https://www.facebook.com/n/?login_alerts%2Fsettings%2F&amp;medium=email&amp;mid=55d15436cG2bf&amp;bcode=2.0964.AbyhsJFgi0Pjg&amp;n_m=redacted%40gmail.com">
                        <center>
                            <font size="3"><span style="font-family:Helvetica Neue,Helvetica,Lucida Grande,tahoma,verdana,arial,sans-serif;white-space:nowrap;font-weight:bold;vertical-align:middle;color:#525252;text-shadow:0 1px 0 #ffffff;font-size:14px;line-height:14px;">Gérer&nbsp;les&nbsp;alertes</span></font>
                        </center>
                    </a>
                </td>
            </tr>
        </table>
    </a>
</td>
```

**Mailspring: via `window.__message.body`**

```
<td class="mb_blk" style=""><a href="https://www.facebook.com/n/?login_alerts%2Fsettings%2F&amp;medium=email&amp;mid=55d15436cG2bf&amp;bcode=2.0964.AbyhsJFgi0Pjg&amp;
n_m=redacted%40gmail.com" style="color:#3b5998;text-decoration:none;"></a>
  <table border="0" width="100%" cellspacing="0" cellpadding="0" style="border-collapse:collapse;">
    <tr>
      <td style="border-collapse:collapse;border-radius:2px;text-align:center;display:block;border:solid 1px #c9ccd1;box-shadow:0 1px 0 rgba(0, 0, 0, 0.04);background:#f6f7f8;
padding:7px 16px 11px 16px;">
        <a href="https://www.facebook.com/n/?login_alerts%2Fsettings%2F&amp;medium=email&amp;mid=55d15436cG2bf&amp;bcode=2.0964.AbyhsJFgi0Pjg&amp;n_m=redacted%40gmail.com" style
="color:#3b5998;text-decoration:none;display:block;"></a>
        <center>
          <font size="3"><span style="font-family:Helvetica Neue,Helvetica,Lucida Grande,tahoma,verdana,arial,sans-serif;white-space:nowrap;font-weight:bold;vertical-align:
middle;color:#525252;text-shadow:0 1px 0 #ffffff;font-size:14px;line-height:14px;">
alertes</span></font>
        </center>
      </td>
    </tr>
  </table>
</td>
```

**Mailspring iframe Dom**
```
<td style=""><a href="https://www.facebook.com/n/?login_alerts%2Fstart%2F&amp;fbid=10155102646468106&amp;s=e&amp;medium=email&amp;mid=55d15042c409aG1dfee161G55d154dc2436cG2bf&amp;bcode=2.1509720964.AbyhsKwbTaEJFgi0Pjg&amp;n_m=couture.lm%40gmail.com" style="color:#3b5998;text-decoration:none;" title="https://www.facebook.com/n/?login_alerts%2Fstart%2F&amp;fbid=10155102646468106&amp;s=e&amp;medium=email&amp;mid=55d15042c409aG1dfee161G55d154dc2436cG2bf&amp;bcode=2.1509720964.AbyhsKwbTaEJFgi0Pjg&amp;n_m=couture.lm%40gmail.com"></a>
    <table border="0" width="100%" cellspacing="0" cellpadding="0" style="border-collapse:collapse;">
        <tbody>
            <tr>
                <td style="border-collapse:collapse;border-radius:2px;text-align:center;display:block;border:solid 1px #344c80;box-shadow:inset 0 1px 1px rgba(255, 255, 255, 0.1),0 1px 0 rgba(0, 0, 0, 0.1);background:#4c649b;padding:7px 16px 11px 16px;">
                    <a href="https://www.facebook.com/n/?login_alerts%2Fstart%2F&amp;fbid=10155102646468106&amp;s=e&amp;medium=email&amp;mid=55d15042c409aG1dfee161G55d154dc2436cG2bf&amp;bcode=2.1509720964.AbyhsKwbTaEJFgi0Pjg&amp;n_m=couture.lm%40gmail.com" style="color:#3b5998;text-decoration:none;display:block;" title="https://www.facebook.com/n/?login_alerts%2Fstart%2F&amp;fbid=10155102646468106&amp;s=e&amp;medium=email&amp;mid=55d15042c409aG1dfee161G55d154dc2436cG2bf&amp;bcode=2.1509720964.AbyhsKwbTaEJFgi0Pjg&amp;n_m=couture.lm%40gmail.com"></a>
                    <span style="font-family:Helvetica Neue,Helvetica,Lucida Grande,tahoma,verdana,arial,sans-serif;white-space:nowrap;font-weight:bold;vertical-align:middle;color:#ffffff;text-shadow:0 -1px 0 #415686;font-size:14px;line-height:14px;">
Afficher&nbsp;la&nbsp;connexion</span>
                </td>
            </tr>
        </tbody>
    </table>
</td>
```

There is a `<a>` tag wrapping the whole table. but in the Mailspring version, the `<a>` closes just before the table instead of wrapping it. Then inside that, there is another `<a>` that's wrapping the `<center>` tag, but it also get closed before wrapping in Mailspring.

**Comment by bengotow** [comment]
Ahh thanks that's super helpful! It looks like there's an HTML sanitizer that is running as part of the underlying `mailcore2` framework that Mailspring uses, and it's using a strict version of the HTML spec where only inline elements are allowed to be within an `<a>` tag. Will see if I can get this fixed.

**Comment by bengotow** [comment]
Hey! This should be further fixed by 3cdc3c27c6c3570b920bb45ad5a4c8d8237c94d7—we were using `html-sanitize`, a node module that allows us to clean things like <object> and <video> tags out of HTML emails. I found a few emails it reliably failed to process and mangled badly, so I re-wrote it to use Chromium's parser. I think this should definitely be resolved now since there are fewer things doing (deep down, probably regex-based) string processing of the message bodies now.
